frontend/src/diagram/DiagramWebSocketContainer.tsx

```diff
@@ -515,10 +515,11 @@ export const DiagramWebSocketContainer = ({ representationId, selection, setSele
     };
     let invokeLabelEditFromContextualPalette;
     if (renameable) {
+      let elementIdSuffix = element instanceof SEdge ? '_centerlabel' : '_label';
       invokeLabelEditFromContextualPalette = () =>
         diagramServer.actionDispatcher.dispatchAll([
           { kind: HIDE_CONTEXTUAL_TOOLBAR_ACTION },
-          new EditLabelAction(element.id + '_label'),
+          new EditLabelAction(element.id + elementIdSuffix),
```

Comment:
Why not give the id of the element, its kind and its label kind? This way, you wouldn't have to try to guess the same information on the backend. You would not look for edges in the list of nodes.

Reply:
I am not sure to understand. If I modify the id, won't it break some other thing (maybe applying a tool) if I change it?

Reply:
@sbegaudeau I'm not sure I understand either. `EditLabelAction` is from Sprotty, we do not control what we pass to its constructor:
```js
export class EditLabelAction implements Action {
    static KIND = 'EditLabel';
    kind = EditLabelAction.KIND;
    constructor(readonly labelId: string) { }
}
```

Reply:
My undestranding is that @sbegaudeau would like for the `editLabelMutation` call to tell the backend directly which kind of label is being edited (node or edge). So instead of passing it:
```js
      const input = {
        projectId: id,
        representationId,
        labelId,
        newText,
      };
```
we would pass somehting like:
```js
      const input = {
        projectId: id,
        representationId,
        elementId,
        elemenKind, // "node" or "edge"
        labelKind, // "center", "start", "end" or any other if/when we support multiple labels for a given diagram element
        newText,
      };
```

This would mean the backend does not have to try and guess what is being edited (as in the current patch, where we try "is there a node with this id? there is not, then is there an edge?").

However, `EditLabelAction` is not under our control, it's the Sprotty specific action which triggers the interaction on the front (open the text field), but when the new text is validated, it's the `ApplyLabelEditAction.KIND` that gets called by Sprotty on our `WebSocketDiagramServer`, which then calls the registered `editLabel` callback (which ends up calling the mutation). The action we get from Sprooty in `handleApplyLabelEditAction` contains the full id, with the embeded suffix.

So either we override the first part of the direct edit protocol to pass a long a more structured object, or we do the label id "parsing" (into `{ elementId, elementKind, labelKind }` on the frontend (but is that really that much cleaner than doing it on the back?).

Or we leave it like that, at least for the moment.